src/main/java/org/sefglobal/scholarx/service/MenteeService.java

```diff
@@ -127,21 +135,38 @@ public Mentee updateAssignedMentor(long menteeId, Long mentorId)
             log.error(msg);
             throw new ResourceNotFoundException(msg);
         }
-        
+
+        Mentor previouslyAssignedMentor = optionalMentee.get().getAssignedMentor();
+        int menteeCountCurrentMentor = optionalMentor.get().getNoOfAssignedMentees();
+        int menteeCountPreviousMentor = 0;
+        if (previouslyAssignedMentor != null) {
+            menteeCountPreviousMentor = previouslyAssignedMentor.getNoOfAssignedMentees();
+        }
         ProgramState programState = optionalMentee.get().getProgram().getState();
         if (programState.equals(ProgramState.ADMIN_MENTEE_FILTRATION)) {
             optionalMentee.get().setAssignedMentor(optionalMentor.get());
             optionalMentee.get().setState(EnrolmentState.ASSIGNED);
+            menteeCountCurrentMentor++;
+            menteeCountPreviousMentor--;
```

Comment:
We can move these 2 lines (with 154,155 lines) to line 162 (below the program state if-else statement)

Reply:
But wouldn't that update the values even if this piece of code executes the `else` block

Reply:
No, the `else` block throws the exception and exits the method

Reply:
Ah yes, didn't notice it 
I'll update the PR, thanks :)

Reply:
Resolved with the commit f8e16d0, Please review